File: Assets/SoundFX/ClipSwitch.cs
using UnityEngine;
using System.Collections;

public class ClipSwitch : MonoBehaviour {

	public AudioClip [] myAudioClip;
	public AudioSource PlaySource;
	private int currentClip = -1;
	
	void Start(){
	
		//myAudioClip = new AudioClip[10];

	}
	
	// Update is called once per frame
	public void NextClip () {
	    currentClip++;

        if (currentClip >= myAudioClip.Length)
			currentClip = 0;
			
		PlaySource.clip = (myAudioClip[currentClip]);
		PlaySource.Play();
	}
}

File: Assets/SoundFX/Clipswitch.js
#pragma strict
var myAudioClip: AudioClip[];

private var currentClip : int = 0;

 

function Start () {

   // LoopClips();

}



function NextClip() {

        GetComponent.<AudioSource>().clip = myAudioClip[currentClip];

        GetComponent.<AudioSource>().Play();
 

        yield WaitForSeconds (GetComponent.<AudioSource>().clip.length);


        currentClip++;

        if (currentClip >= myAudioClip.length)

            currentClip = 0;

}